### PR TITLE
Add css for section metadata style grid width 6 for mobile and tablet

### PR DESCRIPTION
**Pull request by Ruchika4**
* Add padding in mobile and tablet view when Grid Width 6 is added in section metadata 
* This css will only be applicable if section metadata has icon block

Resolves: [MWPW-136028](https://jira.corp.adobe.com/browse/MWPW-136028)

**Test URLs:**

Milo:
- Before: https://main--milo--adobecom.hlx.page/drafts/ruchika/section-metadata?martech=off
- After: https://section-metadata--milo--ruchika4.hlx.page/drafts/ruchika/section-metadata?martech=off

CC:
- Before: https://stage--cc--adobecom.hlx.live/creativecloud/tools/logo-software?martech=off
- After: https://stage--cc--adobecom.hlx.live/creativecloud/tools/logo-software?martech=off&milolibs=section-metadata--milo--ruchika4

Note: Please open the page in mobile and tablet view to test it and look at Ai icon section shared in the screenshot below. Attaching the expected layout which has been approved by our product owners

Screen Shot 2023-09-06 at 10 45 57 AM
Screen Shot 2023-09-06 at 10 46 15 AM

### CODE REVIEW

**Comment by Ruchika4** [comment]
Tagging @Blainegunn @meganthecoder also to confirm if this update will impact negatively on any of their live pages.
Any page with Grid Width 6 in section metadata style will be impacted in mobile and tablet view.

**Comment by Ruchika4** [comment]
I checked for DC and there is only 1 actual page that is using Grid Width 6 [here](https://adobe.sharepoint.com/sites/adobecom/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2Fadobecom%2FShared%20Documents%2Fdc%2Facrobat%2Fguides%2Ffree%2Dbudget%2Dtemplates%2Edocx&viewid=d776cf70%2D9b7e%2D4ab7%2Db9da%2D9e0f8e03a7d2&q=%22Grid%20Width%206%22&parent=%2Fsites%2Fadobecom%2FShared%20Documents%2Fdc&parentview=7) 

I have added restriction for icon block now so this page won't be impacted.

For Bacom also I checked few pages that are using Grid Width 6 like here https://main--bacom--adobecom.hlx.page/products/magento/magento-commerce?milolibs=section-metadata--milo--ruchika4 and they are not impacted post this PR.

cc @Blainegunn  @meganthecoder

**Review by Blainegunn — APPROVED** [review]
tysm!

File: libs/blocks/section-metadata/section-metadata.css [hunk 1]
```diff
@@ -186,6 +186,17 @@ main>.section[class*='-up']>.content {
   .section.five-up {
     grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
   }
+  
+  .section.grid-width-6:has(.icon-block) {
```
Comment (meganthecoder) [inline libs/blocks/section-metadata/section-metadata.css:190]: I don't believe `:has` is supported in all browsers

File: libs/blocks/section-metadata/section-metadata.css [hunk 2]
```diff
@@ -186,6 +186,17 @@ main>.section[class*='-up']>.content {
...
+    padding-left: var(--grid-margins-width-6);
+    padding-right: var(--grid-margins-width-6);
+  }
+}
+
+@media(max-width: 600px){
```
Comment (meganthecoder) [inline libs/blocks/section-metadata/section-metadata.css:196]: Why do you need a media query for mobile? We have a mobile-first setup

**Review by meganthecoder — CHANGES_REQUESTED** [review]
I question whether we want to set a precedent of customizing the grid based on what blocks are inside of it. Does this change come from consonant or is this something that should perhaps be in your consumer repo?

Also, as I put in the comments, I don't think `:has` will pass our testing matrix since it's behind a feature flag in Firefox: https://developer.mozilla.org/en-US/docs/Web/CSS/:has#browser_compatibility

**Comment by Ruchika4** [comment]
> I question whether we want to set a precedent of customizing the grid based on what blocks are inside of it. Does this change come from consonant or is this something that should perhaps be in your consumer repo?
> 
> Also, as I put in the comments, I don't think `:has` will pass our testing matrix since it's behind a feature flag in Firefox: https://developer.mozilla.org/en-US/docs/Web/CSS/:has#browser_compatibility

Thanks for the feedback @meganthecoder . I'll incorporate your feedback and move the css to consumer repo. Closing this PR

**Pull request closed without merging**